src/components/Timeline/PTimelineItem.vue

```diff
@@ -0,0 +1,236 @@
+        'p-timeline-item__left-container--align-right': stacked && align.value === 'right',
+        'p-timeline-item__left-container--align-center': stacked && align.value === 'center',
+        'p-timeline-item__left-container--align-left': stacked && align.value === 'left',
+      },
+      rightContainer: {
+        'p-timeline-item__right-container--layout-stacked': stacked,
+        'p-timeline-item__right-container--align-right': stacked && align.value === 'right',
+        'p-timeline-item__right-container--align-center': stacked && align.value === 'center',
+        'p-timeline-item__right-container--align-left': stacked && align.value === 'left',
+      },
+    }
+  })
+</script>
+
+<style>
+.p-timeline-item {
+  --p-timeline-item-gap: 1rem;
+}
+
+.p-timeline-item { @apply
+  flex
+  items-start
+  relative
+  py-[var(--p-timeline-item-gap)]
+  box-content
+}
+
+.p-timeline-item--layout-stacked { @apply
+  py-[var(--p-timeline-item-gap)]
+}
+
+.p-timeline-item__right-container,
+.p-timeline-item__left-container { @apply
+  self-center
+  flex-1
+  z-[1]
+}
+
+.p-timeline-item__left-container { @apply
+  text-right
+}
+
+.p-timeline-item__right-container { @apply
+  text-left
+}
+
+.p-timeline-item__center-container { @apply
+  flex
+  text-center
+  self-start
+  justify-center
+}
+
+.p-timeline-item__center-container::before { @apply
+  absolute
+  top-0
+  left-1/2
+  bottom-0
+  w-px
+  -translate-x-1/2
+  bg-foreground-200
+  dark:bg-foreground-300
+  z-0
+  ;
+
+  content: '';
+}
+
+.p-timeline-item--layout-alternate:nth-child(odd) { @apply
+  flex-row-reverse
+}
+
+.p-timeline-item--layout-alternate:nth-child(even) { @apply
+  flex-row
+}
+
+.p-timeline-item--layout-default { @apply
+  flex-row
+}
+
+.p-timeline-item--layout-stacked { @apply
+  grid
+}
+
+.p-timeline-item--align-right,
+.p-timeline-item__left-container--align-right,
+.p-timeline-item__right-container--align-right,
+.p-timeline-item--layout-alternate:nth-child(odd) .p-timeline-item__right-container { @apply
+  text-right
+  justify-end
+}
+
+.p-timeline-item--align-center,
+.p-timeline-item__left-container--align-center,
+.p-timeline-item__right-container--align-center { @apply
+  text-center
+  justify-center
+}
+
+.p-timeline-item--align-left,
+.p-timeline-item__left-container--align-left,
+.p-timeline-item__right-container--align-left,
+.p-timeline-item--layout-alternate:nth-child(odd) .p-timeline-item__left-container { @apply
+  text-left
+  justify-start
+}
+
+.p-timeline-item__center-container--layout-stacked { @apply
+  relative
+  h-full
+  items-start
+}
+
+.p-timeline-item__center-container--align-center { @apply
+  h-auto
+  static
+}
+
+.p-timeline-item__center-container--layout-stacked.p-timeline-item__center-container--align-center { @apply
+  mb-[var(--p-timeline-item-gap)]
+}
+
+.p-timeline-item__center-container--layout-stacked::before { @apply
+  -top-[var(--p-timeline-item-gap)]
+  -bottom-[var(--p-timeline-item-gap)]
+}
```

Comment:
We could but I avoided using gap since it breaks the grid in cases where parts of the grid are visible and other parts aren't (in particular if the first row isn't visible in stacked layouts); using margins + some minor absolute positioning keeps the layout a bit more consistent between items

Reply:
Isn't visible meaning it has no content?

Reply:
I think so - there might be a solution in there with template rows but I couldn't out how to make it work